From: Cosmopod OS Build Team <builds@cosmopod.invalid>
Date: Thu, 23 Jul 2026 23:35:00 +0100
Subject: [PATCH] readelf: keep search results const

Newer glibc headers preserve const through strrchr and memchr.  These
pointers only walk read-only DWARF input, so retain the qualifier.

Upstream-Status: Pending

---
 src/readelf.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/readelf.c b/src/readelf.c
index cfb04e4..744b153 100644
--- a/src/readelf.c
+++ b/src/readelf.c
@@ -7970,7 +7970,7 @@ print_debug_units (Dwfl_Module *dwflmod, Dwarf *dbg, Ebl *ebl,
 		    valuestr = dwarf_filesrc (files, num, NULL, NULL);
 		    if (valuestr != NULL)
 		      {
-			char *filename = strrchr (valuestr, '/');
+			const char *filename = strrchr (valuestr, '/');
 			if (filename != NULL)
 			  valuestr = filename + 1;
 		      }
@@ -8724,7 +8724,7 @@ print_form_data (Dwarf *dbg, int form, const unsigned char *readp,
 		 Dwarf_Off str_offsets_base)
 {
   Dwarf_Word val;
-  unsigned char *endp;
+  const unsigned char *endp;
   Elf_Data *data;
   char *str;
   switch (form)
@@ -9216,7 +9216,7 @@ print_debug_line_section (Dwfl_Module *dwflmod __attribute__ ((unused)),
 	{
 	  while (linep < lineendp && *linep != 0)
 	    {
-	      unsigned char *endp = memchr (linep, '\0', lineendp - linep);
+	      const unsigned char *endp = memchr (linep, '\0', lineendp - linep);
 	      if (unlikely (endp == NULL))
 		goto invalid_unit;
 
@@ -9450,7 +9450,7 @@ print_debug_line_section (Dwfl_Module *dwflmod __attribute__ ((unused)),
 		case DW_LNE_define_file:
 		  {
 		    char *fname = (char *) linep;
-		    unsigned char *endp = memchr (linep, '\0',
+		    const unsigned char *endp = memchr (linep, '\0',
 						  lineendp - linep);
 		    if (unlikely (endp == NULL))
 		      goto invalid_unit;
-- 
2.43.0
